applications/tari_base_node/src/bootstrap.rs

```diff
@@ -93,15 +93,15 @@ where B: BlockchainBackend + 'static
             .map(|s| SeedPeer::from_str(s))
             .map(|r| r.map(Peer::from).map(|p| p.node_id))
             .collect::<Result<Vec<_>, _>>()
-            .map_err(|e| ExitError::new(ExitCode::ConfigError, &e))?;
+            .map_err(|e| ExitError::new(ExitCode::ConfigError, e.to_string()))?;
```

Comment:
`.to_string()` call is not necessary above. `ExitError` already does it:

```rust
impl ExitError {
    pub fn new<T: ToString>(exit_code: ExitCode, details: T) -> Self {
        Self {
            exit_code,
            details: Some(details.to_string()),
        }
    }
}

```

Reply:
This is one of those cases where there doesn't seem to be a good answer. Clippy complains about needless_pass_by_value which is technically correct but makes the ergonomics of ExitError worse (esp. in the case of a double ref `str`). I first decided to simply pass in a String and let the caller convert as needed (often a String was passed in anyway so no conversion/allocation needed) but the static str case is worse e.g. `"my error msg".to_string()`. `T: Into<String>` would remove the `needless_pass_by_ref` issue but none of the error types implement this (though perhaps this is the best compromise). In the end it's such a minor issue and decided to move on to get the rest of the clippy fixes over the line.